solutions/standard/main.tf

```diff
@@ -49,17 +49,67 @@ module "kms" {
           force_delete             = true
         }
       ]
+    },
+    {
+      key_ring_name         = var.cos_key_ring_name
+      existing_key_ring     = false
+      force_delete_key_ring = true
+      keys = [
+        {
+          key_name                 = var.cos_key_name
+          standard_key             = false
+          rotation_interval_month  = 3
+          dual_auth_delete_enabled = false
+          force_delete             = true
+        }
+      ]
     }
   ]
 }
 
+#######################################################################################################################
+# COS
+#######################################################################################################################
+
+locals {
+  cos_kms_key_crn   = var.existing_cos_bucket_name != null ? null : local.existing_kms_root_key_id != null ? local.existing_kms_root_key_id : module.kms[0].keys[format("%s.%s", var.cos_key_ring_name, var.cos_key_name)].crn
+  cos_instance_guid = var.existing_cos_instance_crn != null ? element(split(":", var.existing_cos_instance_crn), length(split(":", var.existing_cos_instance_crn)) - 3) : module.cos.cos_instance_guid
+  cos_bucket_name   = var.existing_cos_bucket_name != null ? var.existing_cos_bucket_name : module.cos.bucket_name
+}
+
+module "cos" {
+  source                              = "terraform-ibm-modules/cos/ibm"
+  version                             = "8.2.0"
+  create_cos_instance                 = var.existing_cos_instance_crn == null ? true : false
+  create_cos_bucket                   = var.existing_cos_bucket_name == null ? true : false
+  add_bucket_name_suffix              = var.add_bucket_name_suffix
+  resource_group_id                   = module.resource_group.resource_group_id
+  region                              = var.cross_region_location != null ? null : var.cos_bucket_region
```

Comment:
I think the logic should use var.region by default, however if someone passes a value for `cos_bucket_region` or `cross_region_location` it should use that instead. But we need to ensure values are not passed for both `cross_region_location` and `cos_bucket_region` - if they are, validation should fail

Reply:
Ensure each of the regional related variable description are clear on this